lib/src/selectDialog.dart

```diff
@@ -131,6 +131,11 @@ class _SelectDialogState<T> extends State<SelectDialog<T>> {
     super.initState();
     _debouncer = Debouncer(delay: widget.searchDelay);
     _selectedItemsNotifier.value = widget.selectedValues;
+    widget.searchFieldProps?.controller?.addListener(() {
```

Comment:
We are passing all searchFieldProps via separate props class.
Just create it an pass your controller. 
I do not see the need to make separate callback to subscribe to controller changes.

Reply:
but if we change text by the controller of searchFieldProps ,such as controller.clear() , then the callback wiil not be called, isn't it?